### DISCUSSION TITLE
Why use unsqueeze() here?

### DISCUSSION THREAD

**Discussion by mklingamsetty** [mrdbourke/pytorch-deep-learning#1153] [Q&A]
<img width="431" alt="Question" src="https://github.com/user-attachments/assets/9db6efde-b0dd-4c9e-b3b4-6e8f5b8b1b2b" />

In this image, I dont understand why we have to use the unsqueeze method to make the set from a 1D tensor to a 2D tensor. When I ran my code without the unsqueeze method it still worked, the only difference being the output was a 1D tensor.

**Comment by Prezzo-K** [comment]
It is because most of the time you will work with data with 2 or more dimensions. In that case, it won't make sense if you have your data in one 1D i.e [x1, x2, x3, x4 ...].

But with unsqueeze, the same data is represented as 
[
                                                                                          [x1, x2, ...],  - sample 1
                                                                                          [x1, x2,...]   - sample 2
                                                                                          ....
                                                                                          ]
and u can add more features to each sample.

Hope this helps :)

**Reply by ErsiZhao** [reply]
because we want to have them in vectors for later calculations,  or else just bunch of number and we cant do forward or backward calculation.
